### ISSUE TITLE
Allow arbitrary header names

### ISSUE THREAD

**Issue by passcod** [http-rs/tide#422]
Currently the set and append header methods on Response take a static str for the key:

```rust
pub fn set_header(self, key: &'static str, value: impl AsRef<str>) -> Self;
pub fn append_header(self, key: &'static str, value: impl AsRef<str>) -> Self;
```

I'd like to be able to specify arbitrary header names:

```rust
pub fn set_header(self, key: &str, value: impl AsRef<str>) -> Self;
pub fn append_header(self, key: &str, value: impl AsRef<str>) -> Self;
```

Example usecases:
- echo server, copying (most) headers
- varying header name based on configuration
- (my immediate usecase) processing the request in a 'plugin'

**Comment by passcod** [comment]
For reference, the current workaround:

```rust
let mut res = tide::http::Response::builder();
res.status(meta.status);
for (key, value) in meta.headers {
    res.header(key.as_slice(), value.as_slice());
}

let res = tide::Response::from(res.body(http_service::Body::empty())?);
```

**Comment by Fishrock123** [comment]
This has been fixed since #414 (0.7.0). The header apis now use [`http_types::headers::HeaderName`](https://docs.rs/http-types/1.2.0/http_types/headers/struct.HeaderName.html), which uses `String` internally.
